src/data/matrix/basic.lean

```diff
@@ -251,22 +313,50 @@ by simp_rw [dot_product, mul_comm]
   dot_product v w = v ⟨⟩ * w ⟨⟩ :=
 by simp [dot_product]
 
-@[simp] lemma dot_product_zero [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product v 0 = 0 :=
+section dot_product_zero
+
+variables [non_unital_non_assoc_semiring α] (v : m → α)
+
+@[simp] lemma dot_product_zero  : dot_product v 0 = 0 :=
 by simp [dot_product]
 
-@[simp] lemma dot_product_zero' [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product v (λ _, 0) = 0 :=
+@[simp] lemma dot_product_zero'  : dot_product v (λ _, 0) = 0 :=
 dot_product_zero v
 
-@[simp] lemma zero_dot_product [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product 0 v = 0 :=
+@[simp] lemma zero_dot_product  : dot_product 0 v = 0 :=
 by simp [dot_product]
 
-@[simp] lemma zero_dot_product' [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product (λ _, (0 : α)) v = 0 :=
+@[simp] lemma zero_dot_product' : dot_product (λ _, (0 : α)) v = 0 :=
 zero_dot_product v
 
+end dot_product_zero
+
+section dot_product_one
+
+def vec_one [has_one α] : n → α := λ x, 1
```

Comment:
We already have this, it's spelt 
```suggestion
def vec_one [has_one α] : n → α := 1
```
so there's no point in this definition

Reply:
How do I know which line has over 100 chars? @eric-wieser

Reply:
My above comment still applies, desipite github showing it as outdated.

Reply:
So eric, do we still need the def `vec_one`?

Reply:
I deleted this def.

Reply:
No we don't, you can use `1` everywhere you have currently written `vec_one`